src/app/wallet/receive/receive.component.ts

```diff
@@ -21,77 +27,194 @@ export class ReceiveComponent implements OnInit {
       readable: ['Empty']
     }
 
-  addresses: any = {
-    private: [this.defaultAddress],
-    public: [this.defaultAddress]
-  }
-
   selected: any = {
       id: 0,
       label: 'Empty label',
       address: 'Empty address',
       balance: 0,
       readable: ['empty']
-    };
+  };
 
-    initialized: boolean = false;
+  qrSize: number = 380;
 
-  query: string;
-  searchSubset: any = [];
+  /*
+    UI Pagination
+  */
+  addresses: any = {
+    private: [this.defaultAddress],
+    public: [this.defaultAddress],
+    query: [this.defaultAddress]
 
+  }
+
+  MAX_ADDRESSES_PER_PAGE: number = 6;
   page: number = 1;
-  pages: any = [];
-  nav: any;
 
-  qr: any = {
-    el: undefined,
-    size: undefined
-  }
+  /*
+    initialized boolean: when true => checkFreshAddress is already looping!
+  */
+  initialized: boolean = false;
+
+  /*
+    General
+  */
 
   log: any = Log.create('receive.component');
 
   constructor(private rpc: RPCService) { }
 
+  ngOnInit() {
+
+    // start rpc
+    this.rpc_update();
+  }
+
+/**
+ * Returns the addresses to display in the UI with regards to both pagination and search/query.
+ * Does _NOT_ return the fresh address!
+ * @returns      Object[]
+ */
+  getSinglePage(): Array<Object> {
+    if (this.inSearchMode()) { // in search mode
+      this.addresses['query'] = this.addresses[this.type].filter(el => {
+        return (
+          el.label.toLowerCase().indexOf(this.query.toLowerCase()) !== -1
+          || el.address.toLowerCase().indexOf(this.query.toLowerCase()) !== -1
+        );
+      });
+
+      return this.addresses['query'].slice(((this.page - 1) * this.MAX_ADDRESSES_PER_PAGE), this.page * this.MAX_ADDRESSES_PER_PAGE);
+
+    } else { // not in seach mode
+      return this.addresses[this.type].slice(1 + ((this.page - 1) * this.MAX_ADDRESSES_PER_PAGE), this.page * this.MAX_ADDRESSES_PER_PAGE);
+    }
+  }
+
+/**
+ * Returns the fresh addresses to display in the UI.
+ * @returns      Object[]
+ */
+  getFreshAddress(): Object {
```

Comment:
Can we call this getUnusedAddress, and try to stick to more general terms? :1st_place_medal: